Show Zomic Axis Name and Number in Strut Properties Dialog. Fix Zomic EOF bug in comments. Fix null refs and out-of-bounds array index.

david-hall opened the pull request:
* [Feature] Strut Properties dialog now show the Zomic axis name and
index number when using icosa symmetry system (e.g. Blue -3) unless they
have non-standard (unnamed) directions.

* [Bug Fix] Incorrect use of null when displaying properties of a strut
with non-standard axis.
* [Bug Fix] Incorrect use of null when selecting all struts similar to
one with a non-standard axis.
* [Bug Fix] Incorrect use of null when selecting strut directions "used
in model" while model is empty and the current collection of available
strut directions is also empty.
* [Bug Fix] Permutation.java:81 now avoids an out of bounds condition by
returning NO_ROTATION if index parameter exactly equals m_map.length.
Changed operator from '>' to '>='.

* [Bug Fix] Zomic Single Line Comments encountering EOF before \n or \r
freeze the UI for 5-30 seconds then throw an OutOfMemoryError in
ZomicLexer.
* [Bug Fix] Zomic Multi Line Comments with missing or incomplete '*/'
freeze the UI for 5-30 seconds then throw an OutOfMemoryError in
ZomicLexer.
* [Bug Fix] ZomodLexer SL_COMMENT and ML_COMMENT code is virtually
identical to ZomicLexer, so the exact same fixes were added there but
have not yet been regression tested since there is no Zomod UI.

* [Refactor] ZomicLexer and ZomodLexer and some related code is auto
generated with Antlr based on the corresponding Zomic.g and Zomod.g
source files. The source ".g" files should be tweaked so that the EOF
fixes in the java files are maintained in future rebuilds. As part of
the bug fixes mentioned above, I have already changed the charVocabulary
upper limit from '\uffff' to '\ufffe' in the ".g" files. This is
necessary because '\uffff' is also EOF_CHAR (the end-of file indicator).
The newly added code 'case EOF_CHAR: { break; }' at line 522 of
ZomicLexer.java needs to be added back into Zomic.g as well, but I don't
know the proper syntax to do that in the .g files and I also haven't yet
installed Antler, so I'll leave that step for Scott to complete. Exact
same issue exists with Zomod.g.

vorth commented:
David, I have replaced the vzome-core Maven build with a Gradle build.  It handles the Java3d downloads automatically, and I did the same in vzome-desktop.  I've committed to both projects.  Please confirm that the Gradle builds both work for you.  You should be able to remove your local Maven repo to test the builds.

I updated both README files, too.  Please verify that the new content makes sense to you.

david-hall commented:
OK. I'll check that out too and be sure it all works on Windows before I
work on the ANTLR upgrade.

On Tue, Mar 10, 2015 at 2:48 AM, Scott Vorthmann <notifications@github.com>
wrote:

> David, I have replaced the vzome-core Maven build with a Gradle build. It
> handles the Java3d downloads automatically, and I did the same in
> vzome-desktop. I've committed to both projects. Please confirm that the
> Gradle builds both work for you. You should be able to remove your local
> Maven repo to test the builds.
>
> I updated both README files, too. Please verify that the new content makes
> sense to you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vorth/vzome-core/pull/1#issuecomment-78003884>.
>

david-hall commented:
Scott, 

You've effectively pulled in any of this code by hand that matters. I am closing this pull request so that nobody accidentally tries to merge it.

Dave
